src/EnergyPlus/DaylightingManager.cc

```diff
@@ -5274,18 +5233,19 @@ void GeometryTransformForDaylighting(EnergyPlusData &state)
     NewAspectRatio = 1.0;
 
     CheckForGeometricTransform(state, doTransform, OldAspectRatio, NewAspectRatio);
-    int zoneIndex = 0;
-    for (auto &daylCntrl : state.dataDaylightingData->ZoneDaylight) {
-        zoneIndex++;
-        if (daylCntrl.TotalDaylRefPoints > 0) {
-            auto &zone(state.dataHeatBal->Zone(zoneIndex));
+    // TODO MJW: For now preserve order by zone
+    for (int zoneNum = 1; zoneNum <= state.dataGlobal->NumOfZones; ++zoneNum) {
+        for (int controlNum = 1; controlNum <= state.dataDaylightingData->totDaylightingControls; ++controlNum) {
+            auto &daylCntrl = state.dataDaylightingData->daylightControl(controlNum);
+            if (daylCntrl.zoneIndex != zoneNum) continue;
+            auto &zone(state.dataHeatBal->Zone(daylCntrl.zoneIndex));
```

Comment:
I agree that targets should be specific, not by zone where these objects are. Same for HVAC.